lib/Optimizer/Transforms/ConstPropComplex.cpp

```diff
@@ -0,0 +1,210 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include "PassDetails.h"
+#include "cudaq/Optimizer/Builder/Intrinsics.h"
+#include "cudaq/Optimizer/Dialect/CC/CCOps.h"
+#include "cudaq/Optimizer/Dialect/Quake/QuakeOps.h"
+#include "cudaq/Optimizer/Transforms/Passes.h"
+#include "mlir/Dialect/Complex/IR/Complex.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/Dominance.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "mlir/Transforms/Passes.h"
+
+namespace cudaq::opt {
+#define GEN_PASS_DEF_CONSTPROPCOMPLEX
+#include "cudaq/Optimizer/Transforms/Passes.h.inc"
+} // namespace cudaq::opt
+
+#define DEBUG_TYPE "const-prop-complex"
+
+using namespace mlir;
+
+namespace {
+
+// Replace array ptr casts that throw away the size by a cast to element
+// pointer.
+//
+//%1 = cc.cast %0 : (!cc.ptr<!cc.array<complex<f32> x 4>>) ->
+//! cc.ptr<!cc.array<complex<f32> x ?>>
+// ->
+//%1 = cc.cast %0 : (!cc.ptr<!cc.array<complex<f32> x 4>>) ->
+//! cc.ptr<complex<f32>>
```

Comment:
Looks like clang-format may have formatted this oddly? Could use some whitespace between the `//` and the code here.

nit: we conventionally describe a rewrite using a horizontal line. (See e.g. Peephole.td)

```
      2
    + 1
  ─────
      3
```

Reply:
will update, thanks!

Reply:
removed the pattern